Routes/doctor.js

```diff
@@ -0,0 +1,437 @@
+ *                      type: number                     
+ *          
+ *             
+ */
+
+/**
+ * @swagger
+ * /doctor/newDoctor:
+ *  post:
+ *      summary: Registering a new Doctor  
+ *      description: Add the new doctor data in database
+ *      requestBody:
+ *          required: true
+ *          content:
+ *              application/json:
+ *                  schema:
+ *                      $ref: '#/components/schemas/Doctor'
+ *      responses:
+ *          200:
+ *              description: Added Successfully
+ *              
+ */ 
+
+DoctorRoute.post('/newDoctor',async (req,res) => {
+    const hospital = await Hospital.findById(req.body.hospitalId)
+    if(hospital != null || hospital != undefined){
+
+        const doctor = new Doctors({
+            doctorName: req.body.doctorName,
+            doctorEmail: req.body.doctorEmail,
+            doctorPassword: await bcrypt.hash(req.body.doctorPassword,10),
+            doctorDepartment: req.body.doctorDepartment,
+            hospitalId: req.body.hospitalId,
+            hospitalName: hospital.hospitalName
+            
+        })
+        const newDoctor = await doctor.save()
+        res.json(newDoctor);  
+    }
+    else{
+        res.json({result: "incorrect hospital id try again"})
+    }
+    
+})
+
+/**
+ * @swagger
+ * /doctor/hospital:
+ *  post:
+ *      summary: Registering a new hospital 
+ *      description: Add the new hospital data in database
+ *      requestBody:
+ *          required: true
+ *          content:
+ *              application/json:
+ *                  schema:
+ *                      $ref: '#/components/schemas/Hospital'
+ *      responses:
+ *          200:
+ *              description: Added Successfully
+ *              
+ */ 
+
+DoctorRoute.post('/hospital',verifyToken, async (req,res) =>{
+    const hospital = new Hospital({
+        hospitalName: req.body.hospitalName,
+        hospitalAddress: req.body.hospitalAddress,
+        noOfRooms: req.body.noOfRooms,
+        roomCharge: req.body.roomCharge,
+        
+    })
+    const newhospital = await hospital.save()
+    res.json(newhospital); 
+})
+
+/**
+ * @swagger
+ * /doctor/newMedicine:
+ *  post:
+ *      summary: Registering a new Medicine 
+ *      description: Add the new medicine data in database
+ *      requestBody:
+ *          required: true
+ *          content:
+ *              application/json:
+ *                  schema:
+ *                      $ref: '#/components/schemas/Medicine'
+ *      responses:
+ *          200:
+ *              description: Added Successfully
+ *              
+ */ 
+
+DoctorRoute.post('/newMedicine',async (req,res) => {
+    const medicine = new Medicine({
+        medicineName: req.body.medicineName,
+        medicineType: req.body.medicineType,
+        medicineCost: req.body.medicineCost,
+        medicineDescription: req.body.medicineDescription
+    })
+    const newmedicine = await medicine.save()
+    res.json(newmedicine)
```

Comment:
For any API always remember there are basic 3 steps : 
1. Validation - Always validate you data, that is any type of input that you will get from user
2. Processing - The actual business logic such as the creation of user, searching for data.
3. Output - The format in which the data is returned to user.


Here there is not proper validation for input that is it will accept data even if it's value will be undefined.
The output should be in a proper format, that is the response of the API should be in a single structure such as :
```
{
  status : "Success",
  error : false,
  message : "Everything is good",
  data : { }
}
```

Reply:
validation implemented using [Joi](https://github.com/Sankalp-Sachapara/HospitalPractice/pull/2/files/893ce87fe4fe8a5be2c037597a3a6cfbdffc56ac#r1045073958) 

processing and output [solved](https://github.com/Sankalp-Sachapara/HospitalPractice/pull/2/files/893ce87fe4fe8a5be2c037597a3a6cfbdffc56ac#r1045073575)